Neos.Media/Classes/Domain/Model/AssetCollection.php

```diff
@@ -177,4 +185,33 @@ public function removeTag(Tag $tag)
         }
         return false;
     }
+
+    public function getParent(): ?self
+    {
+        return $this->parent;
+    }
+
+    /**
+     * Throws an error if a circular dependency has been detected
+     * @throws \InvalidArgumentException
+     */
+    private function validateCircularHierarchy(): void
+    {
+        $parent = $this->parent;
+        while ($parent !== null) {
+            $parent = $parent->getParent();
+            if ($parent === $this->parent) {
```

Comment:
```suggestion
            if ($parent !== null && $parent === $this->parent) {
```
Without this null check I get a warning:

![image](https://github.com/neos/neos-development-collection/assets/307571/897eb23b-c766-408f-90a6-d741cac8d7b5)